params/json.libevm_test.go

```diff
@@ -127,20 +128,110 @@ func TestChainConfigJSONRoundTrip(t *testing.T) {
 			t.Cleanup(TestOnlyClearRegisteredExtras)
 			tt.register()
 
-			t.Run("json.Unmarshal()", func(t *testing.T) {
-				got := new(ChainConfig)
-				require.NoError(t, json.Unmarshal([]byte(tt.jsonInput), got))
-				require.Equal(t, tt.want, got)
-			})
+			expectedEncoded := bytes.NewBuffer(nil)
+			err := json.Compact(expectedEncoded, []byte(tt.jsonInput))
+			require.NoError(t, err)
 
-			t.Run("json.Marshal()", func(t *testing.T) {
-				var want bytes.Buffer
-				require.NoError(t, json.Compact(&want, []byte(tt.jsonInput)), "json.Compact()")
+			encoded, err := json.Marshal(tt.want)
+			require.NoError(t, err, "encoding error")
+			require.Equal(t, expectedEncoded.String(), string(encoded))
 
-				got, err := json.Marshal(tt.want)
-				require.NoError(t, err, "json.Marshal()")
-				require.Equal(t, want.String(), string(got))
-			})
+			decoded := new(ChainConfig)
+			err = json.Unmarshal(encoded, decoded)
+			require.NoError(t, err, "decoding error")
+			require.Equal(t, tt.want, decoded)
+		})
+	}
+}
+
+func TestUnmarshalChainConfigJSON(t *testing.T) {
```

Comment:
1. What cases aren't covered by the original tests? (Other than errors, that is)
2. Can the non-erroring cases just be added to the table above?

Re (1); while coverage isn't a perfect metric, I'm seeing full happy-path coverage on your branch with just the originals.

Reply:
See https://github.com/ava-labs/libevm/pull/92#discussion_r1895648853 😉